### PR TITLE
refactor exercise word-search

### PR DESCRIPTION
**Pull request by pheanex**
* rewrite tests so that users have more freedom of implementation
* update example-implementation to fit test-interface
* reduce skeleton to minimal as discussed in #272

### CODE REVIEW

File: exercises/word-search/example.py [hunk 1]
```diff
@@ -1,57 +1,22 @@
...
-        for pos in positions:
-            for d in DIRECTIONS:
-                result = self.find(word, pos, d)
-                if result:
-                    return result
-        return None
+def find_stop(row, column, word, puzzle):
+    directions = [(0, 1), (0, -1), (1, 0), (1, 1), (1, -1), (-1, 0), (-1, 1), (-1, -1)]
+    for d in directions:
+        row_nr, column_nr = row, column
```
Comment (rootulp) [inline exercises/word-search/example.py]: Does `nr` refer to number? If so, why not make this explicit with `row_number` and `column_number`?

Comment (pheanex) [inline exercises/word-search/example.py]: yes it refers to number, i always try to find the sweet spot between overview/compactness and expressiveness ;-). I'll just update it if you think the gain in expressiveness is worth the loss in compactness and brevity

Comment (rootulp) [inline exercises/word-search/example.py]: It's a small nitpick and it's entirely up to you. I was confused by the `_nr` and I'm only pointing it out in the hope that others can avoid similar confusion.

Comment (pheanex) [inline exercises/word-search/example.py]: okay then it absolutely makes sense to change it, thank you for the feedback/review.

File: exercises/word-search/word_search_test.py [hunk 2]
```diff
@@ -1,83 +1,53 @@
...
-                  'oivokprjsm\n'
-                  'pbwasqroua\n'
-                  'rixilelhrs\n'
-                  'wolcqlirpc\n'
-                  'screeaumgr\n'
-                  'alxhpburyi\n'
-                  'jalaycalmp\n'
-                  'clojurermt')
-        self.example = WordSearch(puzzle)
+    puzzle = ['jefblpepre',
```
Comment (behrtam) [inline exercises/word-search/word_search_test.py]: Why did you change the input format? I mean it's only a `split()` away ... just would like to know the reasoning behind this.

File: exercises/word-search/word_search_test.py [hunk 3]
```diff
@@ -1,83 +1,53 @@
...
+              'alxhpburyi',
+              'jalaycalmp',
+              'clojurermt']
 
     def test_horizontal_words_left_to_right(self):
-        self.assertEqual(
-            self.example.search('clojure'),
-            (Point(0, 9), Point(6, 9))
-        )
+        self.assertEqual(search(WordSearchTests.puzzle, 'clojure'), ((9, 0), (9, 6)))
```
Comment (behrtam) [inline exercises/word-search/word_search_test.py]: Why did you reverse the `x, y` order? Normally coordinate systems use `(X, Y)` notation if I'm not mistaken.

**Comment by pheanex** [comment]
Is there any way we can get a third reviewer? @behrtam any suggestions?

**Comment by behrtam** [comment]
Maybe @rootulp has some time to look at it again. (Or do you want a fourth?)
It would help every other reviewer if you could explain your motivations behind the changes I asked you about.

**Comment by pheanex** [comment]
I thought i did explain my motivation (see above). If there is still something that you think i should motivate further, please specify.

**Comment by behrtam** [comment]
I appreciate the effort you have put into improving this exercise from a heavy class based exercise into a lighter flexible function based exercise and hope my maybe annoying questions about some details have not discouraged you.

You switched `(0, 9) -> (9, 0)` so from the typical `(X, Y)` to `(Y, X)` order which is just a small detail, but why? The only explanation I can find ist "freedom of implementation" which for me is not relevant for the `X/Y order` and for the `list vs string input` topic.

**Comment by pheanex** [comment]
No problem.
I tried to say it with "I had to change the order to make it consistent with the naming (row/column)."
and because it is a more straight forward representation to me. That is why i wanted to know how others felt about it. (Hence the question with the o x o x o o x ...).
Also i threw out the newlines (\n) since i thought it is cleaner and simpler without them. (So to just keep whats absolutely necessary and i felt newlines were not absolutely necessary.)

**Comment by behrtam** [comment]
So, the canonical test data is on it's way: https://github.com/exercism/x-common/pull/755

**Comment by behrtam** [comment]
Maybe we could show off `namedtuples` in this exercise?

```python
>>> from collections import namedtuple
>>> Cell = namedtuple('Cell', ['row', 'column'])
>>> start = Cell(2, 2)
>>> end = Cell(row=2, column=12)
```

**Comment by pheanex** [comment]
I guess with canonical test data we can close this
